eclipse-sirius/sirius-components
# [559] Add Selection Dialog on Composite Processor Tool

### Type of this PR 


### Issue(s)

https://github.com/eclipse-sirius/sirius-components/issues/559

### What does this PR do?

First commit of the 'support for a selection dialog on diagram tools'.
When the user selects the Composite Processor Tool in the Flow/Topography Diagram, a 'Selection Dialog' popup is displayed asking for the selection of an object.
The objects are fake for this first commit. The list of objects will be filled with real objects in the following commits associated to this ticket.
The tool is only executed if the user select an object from the list and clicks on the 'Finish' button, otherwise the tool is not executed.

### Screenshot/screencast of this PR

![559-SelectionDialog](https://user-images.githubusercontent.com/717752/122960707-f7699380-d383-11eb-8472-a3cf0af29a74.png)

### How to test this PR?

- [x] Manual Test : please specify

Create a Topography diagram on a Flow model. Then select the 'Composite Processor' creation tool. The 'Selection Dialog' popup should be displayed, allowing to select an object in a list. If you click on 'Cancel' or escape, the popup is closed and nothing happens. If you click on 'Finish', the tool is executed.

### Checklist

- [ ] I have updated the documentation accordingly -> will be done in the following commits of the ticket #559
- [ ] New React components are availables in storybook -> will be done in the following commits of the ticket #559
